src/app/components/code-editor/code-editor.ts

```diff
@@ -1,76 +1,48 @@
-import {Observable} from "rxjs/Observable";
-import {Subscription} from "rxjs/Subscription";
-import {FileModel} from "../../store/models/fs.models";
+import {AbstractCodeEditor} from "../abstract-code-editor/abstract-code-editor";
+import {Subject, Observable, Subscription} from "rxjs";
+import Editor = AceAjax.Editor;
+import Document = AceAjax.Document;
+import IEditSession = AceAjax.IEditSession;
 import Editor = AceAjax.Editor;
 import Document = AceAjax.Document;
 import IEditSession = AceAjax.IEditSession;
-import {WebWorkerService} from "../../services/web-worker/web-worker.service";
-import {ValidationResponse} from "../../services/web-worker/json-schema/json-schema.service";
-import {AbstractCodeEditor} from "../abstract-code-editor/abstract-code-editor";
 
 export class CodeEditor extends AbstractCodeEditor {
 
-    private fileStream: Observable<FileModel>;
+    public contentChanges = new Subject<string>();
 
     private subs: Subscription[] = [];
 
-    private webWorkerService: WebWorkerService;
-
-    public contentChanges: Observable<FileModel>;
-
-    public validationResult: Observable<ValidationResponse>;
-
     constructor(private editor: Editor,
-                private fileStream: Observable<FileModel>,
-                private webWorkerService: WebWorkerService) {
+                private content: Subject<string>,
+                private contentType: Observable<string>,
+                options = {}) {
 
         super();
         this.editor = editor;
+        this.editor.setOptions(options);
 
         // to disable a warning message from ACE about a deprecated method
         this.editor.$blockScrolling = Infinity;
 
-        this.session    = editor.getSession();
-        this.document   = this.session.getDocument();
-        this.fileStream = fileStream;
-
-        this.setTheme('chrome');
+        this.session  = editor.getSession();
+        this.document = this.session.getDocument();
 
-        this.subs.push(
-            this.fileStream.filter(file => file.content !== this.document.getValue())
-                .subscribe(file => {
-                    this.setMode(file.type || ".txt");
-                    this.setText(file.content);
-                    this.validateJsonSchema(file.content);
-                })
-        );
+        this.setTheme("chrome");
 
-        this.contentChanges = Observable.fromEvent(this.editor as any, "change")
-            .debounceTime(300)
-            .map(_ => this.document.getValue())
-            .distinctUntilChanged()
-            .withLatestFrom(this.fileStream, (content, file) => {
-                return Object.assign(file, {content});
-            }).share();
+        this.contentType.subscribe(type => {
+            this.setMode("json");
```

Comment:
Language should not be hardcoded. For SB apps, we should assume type to be JSON, though.